modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SelectVocabularies.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAlert from '@clayui/alert';
+import {ClayDualListBox} from '@clayui/form';
+import React, {useState} from 'react';
+
+export default function ({
+	availableVocabularyNames,
+	currentVocabularyNames,
+	namespace,
+}) {
+	const vocabularies = [currentVocabularyNames, availableVocabularyNames];
+
+	const data = vocabularies.map((inner) =>
+		inner.map(({key, value}) => ({label: value, value: key}))
+	);
```

Comment:
Can there be a large number of vocabularies here? Perhaps this can be expensive since the user will interact a lot with the component and this will be called up after each rendering. Perhaps adding a `useMemo` here can help.

Reply:
👍

Reply:
I don't think you want `useMemo` for this.

Note that `data` is only passed into the `useState` call as an initializer, so you could write this using [the "lazy initial state" form](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state); no need for `useMemo`:

```js
const [items, setItems] = useState(() => {
    return [currentVocabularyNames, availableVocabularyNames].map((inner) =>
        inner.map(({key, value}) => ({label: value, value: key}))
    );
});
```

Reply:
Yes, I hadn't realized it! Good @wincent!